scripts/ci/lint-clang.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC1091
+#
+# Reformats C/C++ code belonging to the repository.
+# The primary Linter with support for reformatting
+# files is clang-format. Any reformatted code causes
+# non-zero exit code (CI failure).
+
+REPOROOT=$(realpath "$(dirname "$0")/../..")
+PATTERN="./*.h ./*.cpp ./*.c"
+
+TPP_CLANGFORMATVER=${TPP_CLANGFORMATVER:-16}
+LINTER=$(command -v "clang-format-${TPP_CLANGFORMATVER}")
+if [ ! "${LINTER}" ]; then
+  LINTER=$(command -v clang-format)
+  if ! ${LINTER} --version | grep -q "${TPP_CLANGFORMATVER}\.[0-9\.]\+"; then
+    echo "ERROR: clang-format v${TPP_CLANGFORMATVER} is missing!"
+    exit 1
+  fi
+fi
+
+if [ "${LINTER}" ]; then
+  COUNT=0
+
+  # LISTFILES="git diff-tree --no-commit-id --name-only HEAD -r --"
+  LISTFILES="git ls-files"
+
+  echo -n "Linting C/C++ files... "
+  cd "${REPOROOT}" || exit 1
+  for FILE in $(eval "${LISTFILES} ${PATTERN}"); do
+    if ${LINTER} -i "${FILE}"; then COUNT=$((COUNT+1)); fi
```

Comment:
It simplifies or unifies all kinds of linters because the failure condition is quite simple: "modified files" in current checkout. Also, this is the way it works for precommit-hook and similar packages. So the point of not reformatting is for local development, right?

Reply:
I can get `-i` back if this makes it more safe for local development.

Reply:
Addressed.